### PR TITLE
xmltopologyreader: add resid tag

### PR DESCRIPTION
**Pull request by junghans**
/cc @razziel89

### CODE REVIEW

**Comment by junghans** [comment]
An update of the manual, somewhere around here:
https://github.com/votca/csg-manual/blob/master/inputfiles.tex#L84
would be nice.

Plus a line here:
https://github.com/votca/csg/blob/master/share/xml/topol.xml#L16
is needed, I guess.

**Comment by RaZziel** [comment]
Ah, you found an old account. I didn't know it was still active. I will use my current account instead and delete this one.

**Comment by junghans** [comment]
@razziel89: Thanks for the documentation.

@MrTheodor: please review.

File: src/libcsg/modules/io/xmltopologyreader.cc [hunk 1]
```diff
@@ -134,25 +136,58 @@ void XMLTopologyReader::ParseMolecule(Property &p, string molname, int nbeads, i
             } catch (runtime_error) {
                 atq = 0.0;
             }
+            try {
+                resid = it->getAttribute<int>("resid");
+                if (resid<=0)
+                    throw std::runtime_error(
```
Comment (jkrajniak) [inline src/libcsg/modules/io/xmltopologyreader.cc]: This runtime_error will be catched by the "catch" below and the code will proceed with resid=-1; Is it an intention?

File: src/libcsg/modules/io/xmltopologyreader.cc [hunk 2]
```diff
@@ -134,25 +136,58 @@ void XMLTopologyReader::ParseMolecule(Property &p, string molname, int nbeads, i
...
+                        "Residue count for beads in topology.molecules.molecule has to be greater than zero");
+            } catch (runtime_error) {
+                resid = -1;
+            }
+            if (!xmlResidues.empty()){
```
Comment (jkrajniak) [inline src/libcsg/modules/io/xmltopologyreader.cc]: Minor, a bit of mixed coding styles, missing spaces before brackets etc..

File: src/libcsg/modules/io/xmltopologyreader.cc (hunk 1 above)
Comment (razziel89) [reply src/libcsg/modules/io/xmltopologyreader.cc]: Thanks for spotting this! It was not intentional and I replaced most of the `std::runtime_error` I added by `std::invalid_argument` as this won't be caught and fits better IMHO. The attached diff `small.txt` should hopefully alleviate this and the coding style issues as well. The attached diff `total.txt` is a diff with respect to the original. Using diffs is quite tedious. Should I create another pull request (I probably should have created one straight away)?

[small.txt](https://github.com/votca/csg/files/940997/small.txt)
[total.txt](https://github.com/votca/csg/files/940999/total.txt)

Comment (jkrajniak) [reply src/libcsg/modules/io/xmltopologyreader.cc]: The easiest way is to commit changes to the branch from where the pull request is made.

Comment (razziel89) [reply src/libcsg/modules/io/xmltopologyreader.cc]: I know this. Unfortunately, I did not create this pull request but @junghans did from a small diff that I posted on the forum.

Comment (junghans) [reply src/libcsg/modules/io/xmltopologyreader.cc]: You can just make another pull request with `xml_residue` being the base branch, but I included the changes quickly this time.

**Comment by junghans** [comment]
@MrTheodor: happy?

File: src/libcsg/modules/io/xmltopologyreader.cc (hunk 1 above)
Comment (razziel89) [reply src/libcsg/modules/io/xmltopologyreader.cc]: Ah, thank you both very much for all your help and patience!

Comment (junghans) [reply src/libcsg/modules/io/xmltopologyreader.cc]: No worries, the first couple of pull requests are the hardest!

**Review by jkrajniak — APPROVED** [review]
LGTM

**Pull request merged**